src/classes/OPP_OpportunityNaming_BATCH.cls

```diff
@@ -77,7 +77,14 @@ public class OPP_OpportunityNaming_BATCH implements Database.Batchable<sObject>,
     * @description Batch process execute method. Names and updates all opportunities in the current batch.
     */
     public void execute(Database.BatchableContext BC, List<Opportunity> oppsToProcess) {
-        lastOppIdProcessed = oppsToProcess[oppsToProcess.size() - 1].Id;
+        // Since execution contexts might not run in order (ex., chunk size 10k,
+        // batch size 2k has 5 batches but they won't run in a guaranteed order)
+        // we first need to check if the last Id is greater than the value in lastOppIdProcessed
+        Id lastIdInScope = oppsToProcess[oppsToProcess.size() - 1].Id;
```

Comment:
Great job with this @seanpat09. The lack of a guaranteed batch sequence is super annoying. Thanks for implementing this in the OpportunityNaming job as well. We'll need to look at CRLP's chunking too in an upcoming work item.